lib/index.d.ts

```diff
@@ -39,14 +40,40 @@ declare namespace Stardog {
         password: string;
     }
 
+    // Kind of a hack, but necessary to get around the way TS libs define the `Request` object.
+    type RequestConstructor = {
+      new (input: string | Request, init?: RequestInit): Request;
+    };
+
+    type RequestCreator<Constructor = RequestConstructor, ReturnType = string | Request> = ({ uri, Request }: { uri: string; Request: Constructor }) => ReturnType;
+
+    /** Optional meta-configuration for a Connection */
+    interface ConnectionMeta {
+        /**
+         * If defined, this method is called whenever a `fetch` call is about
+         * to be made for a given Connection. The method should return a
+         * `Request` object (either a browser `Request` or a `node-fetch`
+         * `Request`) or a string URI, which will be passed as the first
+         * argument to the imminent `fetch` call. This is useful in cases where
+         * fetch needs to allow connections with self-signed certificates, for
+         * example.
+         *
+         * @param {Object} requestData
+         * @param {string} requestData.uri the full URI about to be fetched; includes all URI parts (protocol, hostname, path, query string, etc.)
+         * @param {RequestConstructor | Class<NodeFetchRequest>} requestData.Request a request constructor, conforming either to the browser's Request spec or to `node-fetch`'s Request, depending on environment
+         * @returns {string | Request | NodeFetchRequest} a string URI or a Request object
+         */
+        createRequest?: RequestCreator<RequestConstructor | typeof NodeFetchRequest, string | (Request | NodeFetchRequest)>;
+    }
+
     /** Current version of stardog.js. Maps to package.json */
     export const version: string;
 
     /** Describes the connection to a running Stardog server. */
     export class Connection {
-        constructor(options: ConnectionOptions);
+        constructor(options: ConnectionOptions, meta?: ConnectionMeta);
 
-        config(options: ConnectionOptions): void;
+        config(options: ConnectionOptions, meta?: ConnectionMeta): void;
         headers(): Headers;
```

Comment:
If we're removing the import of Headers, where does this come from? Or was it a duplicate import?

Reply:
The previous import of `Headers` was actually wrong/broken, because it looked like this:

```typescript
import Headers from 'fetch-ponyfill';
```

That doesn't work, because `fetch-ponyfill`'s default export isn't a `Headers` instance at all (it's a function). Unfortunately, `fetch-ponyfill` doesn't export a `Headers` type. In my changed version, at least, the `Headers` type is the usual `Headers` type provided by TypeScript's `lib.es6.d.ts`. Since `fetch-ponyfill` is a ponyfill, its Headers type ought to be structurally the same, and since TypeScript uses duck typing, this should work fine (at least in most cases, if not all).